src/test/scala/mesosphere/marathon/raml/AppConversionTest.scala

```diff
@@ -64,6 +64,11 @@ class AppConversionTest extends UnitTest with ValidationTestLike {
     args = Seq("whatever", "one", "two", "three"),
     container = Some(state.Container.Docker(image = "foo/bla"))
   )
+  private lazy val mesosWithLinuxInfo = AppDefinition(
```

Comment:
We absolutely need more tests (e.g. see `AppsResources` and `PodsResources` classes) for happy and unhappy test cases. This error like missing pod validation would've not gone unnoticed if we had them.

Reply:
we have a large number of case not covered for sure.   I'm not sure this is the time to hold a PR for a higher standard tho... if we don't do more we definitely should capture this as tech debt.